caravel/assets/javascripts/explorev2/components/GroupBy.jsx

```diff
@@ -19,57 +19,43 @@ const defaultProps = {
   groupByColumns: [],
 };
 
-class GroupBy extends React.Component {
-  changeColumns(groupByColumns) {
-    this.props.actions.setGroupByColumns(groupByColumns);
-  }
-  changeMetrics(metrics) {
-    this.props.actions.setMetrics(metrics);
-  }
-  render() {
-    return (
-      <div className="panel space-1">
-        <div className="panel-header">GroupBy</div>
-        <div className="panel-body">
-          <div className="row">
-            <h5 className="section-heading">GroupBy Column</h5>
-            <Select
-              multi
-              name="select-time-column"
-              placeholder="Select groupby columns"
-              options={this.props.groupByColumnOpts}
-              value={this.props.groupByColumns}
-              autosize={false}
-              onChange={this.changeColumns.bind(this)}
-            />
-          </div>
-          <div className="row">
-            <h5 className="section-heading">Metrics</h5>
-            <Select
-              multi
-              name="select-since"
-              placeholder="Select metrics"
-              options={this.props.metricsOpts}
-              value={this.props.metrics}
-              autosize={false}
-              onChange={this.changeMetrics.bind(this)}
-            />
-          </div>
-        </div>
+const GroupBy = (props) => {
+  const selects = [];
```

Comment:
we can assign these select objects directly, rather than using `.push`

also, we could make use of the spread operator here for the common keys for each select.

```
const commonOpts = {
  multi: true,
  width: '12',
};

const selects = [
  {
    key: 'groupByColumns',
    title: 'Group By',
    options: props.groupByColumnOpts,
    value: props.groupByColumns,
    ...commonOpts
 },
 {
    key: 'metrics',
    title: 'Metrics',
    options: props.metricsOpts,
    value: props.metrics,
    ...commonOpts
  },
];
```